portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,10 +27,26 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> messages;
+
+  @Override
+  public void init() {
+    messages = new ArrayList<>();
+    messages.add("Howdy");
+    messages.add("Hey, how are you?");
+    messages.add("Good morning, friend!");
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-   
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Lindsey!");
+    String json = convertToJson(messages);
```

Comment:
Should this be a parameter to this function?

Reply:
Correct me if I'm wrong about any of this, but from my understanding, since "messages" is a global variable, it should be in scope for this function. Also, I believe that the doGet() function is only called when a request is made by the client to the URL, so this might be a built-in function signature (it was provided by the skeleton code).

Reply:
Ahh I missed the global variable above. Thank you.

Yes. The doGet is only executed via a url call and the parameter list is standard. Although, it can be amended. For instance, if you don't want to return a response you can remove that parameter. You can also set this up to be called from a different function somewhere else in the code, as long as you pass the necessary params as with any other function call.

Reply:
That's good to know. I'll definitely play around with it when it comes up in the future.